### PR TITLE
feat: Mouse picking with highlight

### PR DESCRIPTION
**Pull request by PaablooCH**
Once you select an object it will be highlighted (I have choosen the organization color).
I can do highlight also when there is somehing in front of the main object, but as you wish.

### CODE REVIEW

File: Source/DataModels/GameObject/GameObject.cpp [hunk 1]
```diff
@@ -91,6 +92,52 @@ void GameObject::Draw() const
...
+{
+	std::queue<const GameObject*> gameObjectQueue;
+	gameObjectQueue.push(this);
+	while (!gameObjectQueue.empty())
+	{
+		const GameObject* currentGo = gameObjectQueue.front();
+		gameObjectQueue.pop();
+		for (GameObject* child : currentGo->GetChildren())
+		{
+			if (child->IsActive()) {
```
Comment (markmeliaa) [inline Source/DataModels/GameObject/GameObject.cpp:105]: the opening `{` of this `if statement` should go in the next line

File: Source/DataModels/GameObject/GameObject.cpp [hunk 2]
```diff
@@ -91,6 +92,52 @@ void GameObject::Draw() const
...
```
Comment (markmeliaa) [inline Source/DataModels/GameObject/GameObject.cpp:129]: same here with the `{`

**Review by markmeliaa — APPROVED** [review]
Really nice addition Pablo, looks great!

File: Source/DataModels/GameObject/GameObject.cpp [hunk 3]
```diff
@@ -91,6 +92,52 @@ void GameObject::Draw() const
...
+				gameObjectQueue.push(child);
+			}
+		}
+		std::vector<ComponentMeshRenderer*> meshes = 
+			currentGo->GetComponentsByType<ComponentMeshRenderer>(ComponentType::MESHRENDERER);
+		for (ComponentMeshRenderer* mesh : meshes) {
```
[the remark this answers is not in the document]
Comment (PaablooCH) [reply Source/DataModels/GameObject/GameObject.cpp:135]: thank you, I review it and upload it

**Review by xavi1111 — APPROVED** [review]
Tested the feature and seems to be working perfectly. I didn't encounter any bugs. Really nice addition.

**Review by markmeliaa — APPROVED** [review]
Nice catch! Works fine and looks fine :)

**Pull request merged**